src/pani/registry.cljs

```diff
@@ -0,0 +1,18 @@
+(ns pani.registry)
+
+(defonce unsubs (atom {}))
+
+(defn register-listener [ref type unsub!]
+  (swap! unsubs update-in [(str ref) type] #(set (conj % unsub!))))
+
+(defn- flatten-vals [xs]
+  (if-not (map? xs)
+    xs
+    (reduce into (map flatten (vals xs)))))
+
+(defn- call-all! [fs] (doseq [f fs] (f)))
+
+(defn disable-unsubs! [& path]
+  (swap! @unsubs update-in
+         (map str path)
+         (comp call-all! flatten-vals)))
```

Comment:
Naming fail (did find-replace on `listeners` -> `unsubs` for the atom. `disable-listeners` indeed.

Possibly trying to be too concise here, essentially just overloaded to take paths length 0, 1, 2 or 3, which correspond to "remove all callbacks", "remove all callbacks for this ref", "remove all callbacks of this type for this ref" and "remove just this precise callback."

0.0.3 code: https://github.com/verma/pani/blob/master/src/pani/cljs/core.cljs#L10-L39